Schema Docs: usdUI: Create user docs for NodeGraphNodeAPI

beersandrew opened the pull request:
### Description of Change(s)

- Added contents for userDocBrief & userDoc for each component of NodeGraphNodeAPI

### Link to proposal ([if applicable](https://openusd.org/release/contributing_to_usd.html#step-1-get-consensus-for-major-changes))

### Fixes Issue(s)

### Checklist

[ ] I have created this PR based on the dev branch

[ ] I have followed the [coding conventions](https://openusd.org/release/api/_page__coding__guidelines.html)

[ ] I have added unit tests that exercise this functionality (Reference: 
[testing guidelines](https://openusd.org/release/api/_page__testing__guidelines.html))

[ ] I have verified that all unit tests pass with the proposed changes

[x] I have submitted a signed Contributor License Agreement (Reference: 
[Contributor License Agreement instructions](https://openusd.org/release/contributing_to_usd.html#contributor-license-agreement))

beersandrew committed:
feat: Create user docs for NodeGraphNodeAPI

- Added contents for userDocBrief & userDoc for each component of NodeGraphNodeAPI

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+- We expect that the `Preview Surface` node will be red, while the `Color` node will be blue
+- The associated doc links are provided for the user to get more node info
+- `Preview Surface` is `open` and `Color` is `expanded`, the implementer will have input in what `Color` shows but it's expected to reveal more details than `Preview Surface`.
+- The icon's come from different files and are meant to be descriptive for the nodes, not the content of the nodes.
```
icon's -> icons

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+z-index or relative depth strength (`ui:nodegraph:node:stackingOrder`), among others. `NodeGraphNodeAPI` allows users to specify this type of 
+node information.
+
+The example below shows a typical use case for shader nodes with two nodes `Preview Surface` and `Color`. Imagining a shader node graph, we can infer from the below example the following information:
```
Optional: Maybe we could say something like:

"The example below shows a typical use case for a shader node graph with two nodes: 'Preview Surface' and 'Color'. We can infer from the example the following information:

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+The example below shows a typical use case for shader nodes with two nodes `Preview Surface` and `Color`. Imagining a shader node graph, we can infer from the below example the following information:
+- We expect that the `Preview Surface` node will be red, while the `Color` node will be blue
+- The associated doc links are provided for the user to get more node info
+- `Preview Surface` is `open` and `Color` is `expanded`, the implementer will have input in what `Color` shows but it's expected to reveal more details than `Preview Surface`.
```
Optional, but maybe instead of "the implementer", we might say something like "How the expansion state will be presented will depend on the context or tool the nodes are viewed in, but it's expected to....."

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
-            string userDocBrief = """Brief user doc."""
-            string userDoc = """Longer-form user doc."""
+            string userDocBrief = """An x,y coordinate pair expressing the location of the node."""
+            string userDoc = """An x,y coordinate pair expressing the location of the node."""
```
Optional, but in the long-form content it might be worth noting that the position is relative to any parent node's position (if there is a parent node). Also maybe that the Y-coordinate values increase "down" (there's probably a better way to say this). The audience and use-case I'm imagining that might want to know things like this is a TD trying to set up a node graph for their department, doing things directly in USD. Maybe this isn't a common use-case?

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+
+        uniform color3f ui:nodegraph:node:displayColor = (0.0, 0.0, 1.0)
+        uniform string ui:nodegraph:node:docURI = "https://github.com/AcademySoftwareFoundation/MaterialX/blob/main/documents/Specification/MaterialX.Specification.md#procedural-nodes"
+        uniform token ui:nodegraph:node:expansionState = "expanded"
```
Possibly a weird question -- is "expanded" a valid value for expansionState? In the schema definition I only see open, closed, or minimized. But maybe I missed something?

beersandrew replied:
Apologies, I think I saw `minimized` and it somehow transformed into `expanded` in my head when I created the example.

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+            string userDocBrief = """An image to classify the node."""
+            string userDoc = """An image to classify the node.
+
+Meant to help describe the node itself, not the result of the action. For example with an Add node, an icon might be a + sign.
```
I like the  Add node example! Makes perfect sense.

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
+            string userDocBrief = """Optional x,y dimensions for the node."""
+            string userDoc = """Optional x,y dimensions for the node.
+
+This is optional because the size of the node can be generated by it's number of inputs and outputs.
```
it's -> its

dsyu-pixar commented on pxr/usd/usdUI/userDoc/schemaUserDoc.usda:
```diff
@@ -22,56 +22,116 @@ over "GLOBAL" (
-            string userDocBrief = """Brief user doc."""
-            string userDoc = """Longer-form user doc."""
+            string userDocBrief = """A link that points to documentation for the node."""
+            string userDoc = """A link that points to documentation for the node."""
```
Optional, and I'm not sure this is strictly required, but maybe a note that the link is expected to be in URI format?

dsyu-pixar reviewed (commented):
Andy, looks great -- added a couple review comments.

beersandrew committed:
fix: addressing PR comments

- fix various grammatical mistakes
- expanded does not exist, using closed
- Added clarity to some long form docs
